src/coreclr/jit/rangecheck.cpp

```diff
@@ -1185,7 +1177,12 @@ bool RangeCheck::DoesBinOpOverflow(BasicBlock* block, GenTreeOp* binop)
 bool RangeCheck::DoesVarDefOverflow(GenTreeLclVarCommon* lcl)
 {
     LclSsaVarDsc* ssaDef = GetSsaDefAsg(lcl);
-    return (ssaDef == nullptr) || DoesOverflow(ssaDef->GetBlock(), ssaDef->GetAssignment()->gtGetOp2());
+    if (ssaDef == nullptr)
+    {
+        // Parameters don't have definitions but are known to never overflow
+        return !m_pCompiler->lvaIsParameter(lcl->GetLclNum());
+    }
+    return DoesOverflow(ssaDef->GetBlock(), ssaDef->GetAssignment()->gtGetOp2());
```

Comment:
Hmm, it is possible to get there for defs that aren't just `FIRST_SSA_NUM` ones, but also non-`LCL_VAR` ones. I think it would be safer to check that what we are seeing here is really the def that comes from outside the method:

Reply:
@SingleAccretion not sure I understand, can you elaborate?

Reply:
(Sorry for the delay).

The `!m_pCompiler->lvaIsParameter(lcl->GetLclNum());` check permits us to say that for any SSA definition (1st, 2nd, 3rd, etc) of a parameter, we assert it will not overflow.

However, that seems like a dangerous assumption, because we only that is the case for the first definition (one that comes from outside the method), so I was suggesting the check be strengthened to this:
```c
if (ssaDef == nullptr)
{
    // Parameter definitions that come from outside the method could not have overflown.
    if ((lcl->GetSsaNum() == SsaConfig::FIRST_SSA_NUM) && m_pCompiler->lvaIsParameter(lcl->GetLclNum()))
    {
        return false;
    }
    
    return true;
}
```
I believe it is the case today we will not end up here with `ssaDef == nullptr` and `lcl->GetSsaNum() != SsaConfig::FIRST_SSA_NUM`, but that's a fragile (and non-obvious) invariant to rely on.

Reply:
You could also assert in the `return false` clause that the ssa def comes from `fgFirstBB`.